messages/src/bin/dap_decode.rs

```diff
@@ -58,7 +58,7 @@ fn decode_dap_message(message_file: &str, media_type: &MediaType) -> Result<Box<
             if let Ok(decoded) = AggregateShareReq::<TimeInterval>::get_decoded(&message_buf) {
                 Box::new(decoded) as Box<dyn Debug>
             } else {
-                Box::new(AggregateShareReq::<FixedSize>::get_decoded(&message_buf))
+                Box::new(AggregateShareReq::<FixedSize>::get_decoded(&message_buf)?)
                     as Box<dyn Debug>
```

Comment:
I don't think this is right. It works because `Result<AggregateShareReq::<FixedSize>, Error>` must implement `Debug`, but we want to apply `?` here so that we can yield an obvious error should decoding the message fail.

Reply:
My patch is adding the `?` operator, so that this branch matches the `AggregateInitializeReq` branch above. However, I think I'm going to rewrite this with let bindings and ascribed types, before everything gets turned into trait objects, so that it's clearer and we can avoid similar problems with future changes.

Reply:
I'm sorry, I read the diff exactly backward. And I put this bug in the program to begin with.